src/glyphs.cpp

```diff
@@ -5,203 +5,620 @@
+    {
+        FT_Error face_error = FT_New_Face(library, baton->file_name.c_str(), i, &ft_face);
+        if (face_error) {
+            baton->error_name = std::string("could not open Face: '") + baton->file_name + "'";
+            return;
+        }
+
+        llmr::glyphs::fontstack *mutable_fontstack = glyphs.add_stacks();
+        mutable_fontstack->set_name(std::string(ft_face->family_name) + " " + ft_face->style_name);
+        mutable_fontstack->set_range(std::to_string(baton->start) + "-" + std::to_string(baton->end));
+
+        const double scale_factor = 1.0;
+
+        // Set character sizes.
+        double size = 24 * scale_factor;
+        FT_Set_Char_Size(ft_face,0,(FT_F26Dot6)(size * (1<<6)),0,0);
+
+        for (std::vector<uint32_t>::size_type i = 0; i != baton->chars.size(); i++) {
+            FT_ULong char_code = baton->chars[i];
+            glyph_info glyph;
+
+            // Get FreeType face from face_ptr.
+            FT_UInt char_index = FT_Get_Char_Index(ft_face, char_code);
+
+            if (!char_index) continue;
+
+            glyph.glyph_index = char_index;
+            RenderSDF(glyph, 24, 3, 0.25, ft_face);
+
+            // Add glyph to fontstack.
+            llmr::glyphs::glyph *mutable_glyph = mutable_fontstack->add_glyphs();
+            mutable_glyph->set_id(char_code);
+            mutable_glyph->set_width(glyph.width);
+            mutable_glyph->set_height(glyph.height);
+            mutable_glyph->set_left(glyph.left);
+            mutable_glyph->set_top(glyph.top - glyph.ascender);
+            mutable_glyph->set_advance(glyph.advance);
+
+            if (glyph.width > 0) {
+                mutable_glyph->set_bitmap(glyph.bitmap);
+            }
+
+        }
+        if (ft_face) {
+            FT_Done_Face(ft_face);
+        }
+    }
+
+    baton->message = glyphs.SerializeAsString();
+
+    FT_Done_FreeType(library);
 }
 
-NAN_METHOD(Glyphs::Range) {
+void AfterRange(uv_work_t* req) {
     NanScope();
+    RangeBaton* baton = static_cast<RangeBaton*>(req->data);
 
-    // Validate arguments.
-    if (args.Length() < 1 || !args[0]->IsString()) {
-        return NanThrowTypeError("fontstack must be a string");
+    if (!baton->error_name.empty()) {
+        v8::Local<v8::Value> argv[1] = { NanError(baton->error_name.c_str()) };
+        NanMakeCallback(NanGetCurrentContext()->Global(), NanNew(baton->callback), 1, argv);
+    } else {
+        v8::Local<v8::Array> js_faces = NanNew<v8::Array>();
+        unsigned idx = 0;
+        v8::Local<v8::Value> argv[2] = { NanNull(), NanNewBufferHandle(baton->message.data(), baton->message.size()) };
+        NanMakeCallback(NanGetCurrentContext()->Global(), NanNew(baton->callback), 2, argv);
     }
 
-    if (args.Length() < 2 || !args[1]->IsString()) {
-        return NanThrowTypeError("range must be a string");
-    }
+    NanDisposePersistent(baton->callback);
+    delete baton;
+};
+
+struct User {
```

Comment:
Not following how the comment relates or the code referenced above. Maybe form up a separate ticket to help me understand your question?